tensorflow/python/framework/ops.py

```diff
@@ -253,9 +254,24 @@ def disable_tensor_equality():
   Tensor._USE_EQUALITY = False  # pylint: disable=protected-access
 
 
+DataType = TypeVar("DataType",
+                   dtypes.Float16, dtypes.Float32, dtypes.Float64,
```

Comment:
I'm a bit worried that this will cause a lot of duplication in the future. Let's use `TypeVar("DataType", bound=dtypes.DType)` instead.

Reply:
Yeah this sounds good, we can set an upper bound since we are including all subclasses of DType